application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,184 @@
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.Command;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.requests.restaction.MessageCreateAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.utils.DiscordClientAction;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * Implements the /modmail command, which allows users to contact a moderator within the server or
+ * even if not part of the server anymore, e.g. after a ban.
+ * <p>
+ * Therefore, the user uses the SlashCommand /modmail and the message are forwarded to moderators in
+ * a dedicated channel.
+ * <p>
+ * Messages are then forwarded to a dedicated channel, given by
+ * {@link Config#getModAuditLogChannelPattern()}.
+ */
+
+public final class ModMailCommand extends SlashCommandAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(ModMailCommand.class);
+    private static final String COMMAND_NAME = "modmail";
+    private static final String OPTION_MESSAGE = "message";
+    private static final String OPTION_STAY_ANONYMOUS = "stay-anonymous";
+    private static final String OPTION_GUILD = "server";
+    private static final int COOLDOWN_DURATION_VALUE = 30;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private static final Color AMBIENT_COLOR = Color.BLACK;
+    private final Cache<Long, Instant> authorToLastModMailInvocation = createCooldownCache();
+    private final Predicate<String> modMailChannelNamePredicate;
+
+
+    /**
+     * Creates a new instance.
+     *
+     * @param jda the JDA instance to retrieve guildCache
+     * @param config to get the channel to forward modmails to
+     */
+    public ModMailCommand(JDA jda, Config config) {
+        super(COMMAND_NAME, "Contact the moderators of the selected guild",
+                CommandVisibility.GLOBAL);
+
+        OptionData messageOption = new OptionData(OptionType.STRING, OPTION_MESSAGE,
+                "What do you want to tell them?", true);
+        OptionData guildOption = new OptionData(OptionType.STRING, OPTION_GUILD,
+                "The server to contact mods from", true);
+        OptionData anonymousOption = new OptionData(OptionType.BOOLEAN, OPTION_STAY_ANONYMOUS,
+                "If set, your name is hidden - note that mods then can not get " + "back to you",
```

Comment:
why the text split with `+`? looks like a leftover from refactoring

Reply:
It sometimes adds that by it self after build, or when I do spotless with gradle

Reply:
Done